### PR TITLE
Add ability to create users within `GYTxGameMonad`

### PR DESCRIPTION
**Pull request by TotallyNotChase**
This adds a missing feature that used to exist with PSM: the ability to create (and fund) users.

Tested in downstream projects and works great!

### CODE REVIEW

**Comment by sourabhxyz** [comment]
@TotallyNotChase Could you rebase it? Thanks!

**Comment by TotallyNotChase** [comment]
> @TotallyNotChase Could you rebase it? Thanks!

Done!

File: src/GeniusYield/Types/Key.hs [hunk 1]
```diff
@@ -360,6 +363,9 @@ stakeVerificationKeyFromApi = coerce
 stakeVerificationKeyToApi :: GYStakeVerificationKey -> Api.VerificationKey Api.StakeKey
 stakeVerificationKeyToApi = coerce
 
+stakeVerificationKeyToLedger :: GYStakeVerificationKey -> Ledger.VKey r Ledger.StandardCrypto
```
Comment (sourabhxyz) [inline src/GeniusYield/Types/Key.hs:366]: I would have expected `r` to be `Ledger.Staking`, perhaps it doesn't matter.

File: src/GeniusYield/TxBuilder/Class.hs [hunk 2]
```diff
@@ -244,6 +244,14 @@ signTxBodyWithStakeImpl kM txBody = (\(pKey, sKey) -> signGYTxBody txBody $ GYSo
 class (GYTxMonad (TxMonadOf m), GYTxSpecialQueryMonad m) => GYTxGameMonad m where
     -- | Type of the supported 'GYTxMonad' instance that can participate within the "game".
     type TxMonadOf m = (r :: Type -> Type) | r -> m
+    -- TODO: Maybe introduce a user generation config type that this function can take?
```
Comment (sourabhxyz) [inline src/GeniusYield/TxBuilder/Class.hs:247]: Yes, I think we should at a later point, something like `CreateUserConfig` inside `GeniusYield.Test.Privnet.Ctx`.

**Review by sourabhxyz — APPROVED** [review]
Awesome addition @TotallyNotChase! PR LGTM.

**Pull request merged**